solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -197,9 +197,17 @@ private static DirectoryReader getReader(
   private static DirectoryReader wrapReader(SolrCore core, DirectoryReader reader)
       throws IOException {
     assert reader != null;
-    return ExitableDirectoryReader.wrap(
-        UninvertingReader.wrap(reader, core.getLatestSchema().getUninversionMapper()),
-        SolrQueryTimeoutImpl.getInstance());
+    switch (IndexSchema.getUninvertibleSupport()) {
+      case DEFAULT_FALSE:
+      case DEFAULT_TRUE:
+        reader = UninvertingReader.wrap(reader, core.getLatestSchema().getUninversionMapper());
```

Comment:
If the schema has no field types declared UNINVERTABLE (wether it occurs explicitly or via a default), we could skip this wrap here.  Not strictly necessary but it removes a thin layer of indirection.  Such logic could go here or be in the wrap method if, say, we have the uninversionMapper return null.

Reply:
👍  I'll look into this. Maybe worth doing if there's an easy way to track this, but if not I'm not sure it'd be worth the effort (the cost of wrapping presumably being relatively low -- I just did it here because it was easy 🙂).